### PR TITLE
Pioneer DDJ-FLX4 mapping improvements

### PR DESCRIPTION
**Pull request by grizeldi**
2 improvements to the built in FLX4 mapping:

1. Controller now properly exits Bluetooth pairing mode while connected to Mixxx.
2. Removed the mapping for the software headphone gain knob, as the knob on the controller already affects hardware gain, which resulted in very sensitive gain knob (as discussed in the last couple messages of #11245).

The behavior of headphone cueing and mixing now seems to be consistent with how FLX4 works with Rekordbox.

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
Welcome at Mixxx!
As a first-time contributor we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

**Comment by grizeldi** [comment]
Done. Apple complicating things for devs again I see :sweat_smile:

File: res/controllers/Pioneer-DDJ-FLX4-script.js [hunk 1]
```diff
@@ -169,6 +169,11 @@ PioneerDDJFLX4.lights = {
 // Store timer IDs
 PioneerDDJFLX4.timers = {};
 
+// Keep alive timer
+PioneerDDJFLX4.sendKeepAlive = function() {
+    midi.sendSysexMsg([0xF0, 0x00, 0x40, 0x05, 0x00, 0x00, 0x04, 0x05, 0x00, 0x50, 0x02, 0xf7], 12);
+};
```
Comment (Swiftb0y) [inline res/controllers/Pioneer-DDJ-FLX4-script.js:175]: just out of curiosity, how did you find this message? Why is that one the proper one to keep it out of bluetooth mode?

**Review by Swiftb0y — COMMENTED** [review]
LGTM otherwise. Thank you.

File: res/controllers/Pioneer-DDJ-FLX4-script.js (hunk 1 above)
Comment (grizeldi) [reply res/controllers/Pioneer-DDJ-FLX4-script.js:175]: In the official MIDI mapping documentation, no sysex messages are documented, even though the controller seems to employ quite a couple. A while ago I captured a bit of traffic between Rekordbox and my FLX4 with Wireshark for another project and this is the one sysex that had a rather clear purpose, as Rekordbox sends it every 200ms and if it isn't sent in time, the controller reverts into BT pairing mode.

Is that a problem? I rarely have to reverse engineer stuff, so I'm not sure how the legality of it works with international projects.

Comment (Swiftb0y) [reply res/controllers/Pioneer-DDJ-FLX4-script.js:175]: I have no idea about the legal aspects, but [usually reverse-engineering is the only solution for Pioneer hardware](https://swiftb0y.github.io/CDJHidProtocol/hid-analysis/) (I'm the author, so :shrug: :wink: ). I was rather just asking whether there was some documentation out there that I didn't know about. 

Can you just add a quick comment next to the function that the sysex was reverse-engineered from packet captures? That way these questions don't crop up in the future. Thank you.

Comment (grizeldi) [reply res/controllers/Pioneer-DDJ-FLX4-script.js]: To my knowledge, there's only [this](https://www.pioneerdj.com/-/media/pioneerdj/software-info/controller/ddj-flx4/ddj-flx4_midi_message_list_e1.pdf), which doesn't cover any sysex, sadly. Pioneer hardware/software seems to be a lot less extensible than professional grade software I'm used to from other industries.

I added a comment with the notice.

**Review by Swiftb0y — APPROVED** [review]
LGTM. Thank you.

**Comment by Swiftb0y** [comment]
too quick yet again, this should've gone into 2.4... whats the procedure? cherry-pick to backport?

**Comment by daschuer** [comment]
Yes.

**Comment by Swiftb0y** [comment]
thank you. done

**Pull request merged**